### ISSUE TITLE
Not working with HomeAssistant 2024.12.x

### ISSUE THREAD

**Issue by andreynepo** [gritaro/gigachain#11]
Stopped working after upgrade to HA 2024.12.x

**Comment by arivelli** [comment]
It's not working for me neither, I found this error on the log files... maybe related with the issue, maybe not

```
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 269, in async_process_requirements
    await self._async_process_requirements(name, missing)
  File "/usr/src/homeassistant/homeassistant/requirements.py", line 307, in _async_process_requirements
    raise RequirementsNotFound(name, list(failures))
homeassistant.requirements.RequirementsNotFound: Requirements for gigachain not found: ['gigachain==0.1.7.1'].
```

**Comment by gritaro** [comment]
Yes, it stopped working with the latest HA releases due to a dependency issue.
This integration relies on the "gigachain" library, which depends on a "numpy" library version (>=1.0.0,<2.0.0) that conflicts with HA "numpy" (2.1.3).

And as far as I know, "gigachain" has no plans to raise his "nimphy" to 2+.
So for now I have no idea how to fix it :-(

**Comment by andreynepo** [comment]
Move to langchain-gigachat? ))) 
https://github.com/ai-forever/langchain-gigachat/tree/master/libs/gigachat
Not sure if it is the same...

**Comment by MSMikl** [comment]
> Yes, it stopped working with the latest HA releases due to a dependency issue. This integration relies on the "gigachain" library, which depends on a "numpy" library version (>=1.0.0,<2.0.0) that conflicts with HA "numpy" (2.1.3).
> 
> And as far as I know, "gigachain" has no plans to raise his "nimphy" to 2+. So for now I have no idea how to fix it :-(

Only if to fork gigachain and change its dependencies to nimphy 2+ and dependency of HASS integration to forked gigachain

**Comment by gritaro** [comment]
Fixed in https://github.com/gritaro/gigachain/releases/tag/0.1.8

**Comment by kilkams** [comment]
Привет. после обновления на 0.1.8 ошибка изменилась на 
homeassistant.requirements.RequirementsNotFound: Requirements for gigachain not found: ['gigachain@git+https://github.com/gritaro/gigachain-fork@dev#subdirectory=libs/langchain'].

**Comment by gritaro** [comment]
> Привет. после обновления на 0.1.8 ошибка изменилась на homeassistant.requirements.RequirementsNotFound: Requirements for gigachain not found: ['gigachain@git+https://github.com/gritaro/gigachain-fork@dev#subdirectory=libs/langchain'].

ХА запускаете, наверно, не в докер контейнере?
Не совсем понятно - с гитхаба скачать не может, либо на сборке пакета падает, дебаг-логи не помешали бы.
Попробую wheel-пакеты собрать и выложить - может поможет.

**Comment by kilkams** [comment]
> > Привет. после обновления на 0.1.8 ошибка изменилась на homeassistant.requirements.RequirementsNotFound: Requirements for gigachain not found: ['gigachain@git+https://github.com/gritaro/gigachain-fork@dev#subdirectory=libs/langchain'].
> 
> ХА запускаете, наверно, не в докер контейнере? Не совсем понятно - с гитхаба скачать не может, либо на сборке пакета падает, дебаг-логи не помешали бы. Попробую wheel-пакеты собрать и выложить - может поможет.

Сегодня заработало. Возможно, проблемы с доступом к гитхабу были. В последнее время что то периодически блокируется.
